http-gateway/web/src/common/services/websocket-event-client.js

```diff
@@ -0,0 +1,274 @@
+
+      if (!correlationId) {
+        return
+      }
+
+      // Save the subscriptionId to this.idsMap
+      if (
+        !this.idsMap[correlationId] &&
+        operationProcessed?.errorStatus?.code === 'OK'
+      ) {
+        this.idsMap[correlationId] = subscriptionId
+      }
+
+      // Invoke the event listener attached to this correlation id
+      if (
+        this.events?.[correlationId]?.listenerEnabled &&
+        !operationProcessed
+      ) {
+        this.events[correlationId]?.listener(args)
+      }
+    } catch (e) {
+      console.log('Cannot parse JSON from WS', message)
+    }
+  }
+
+  _reSendSubscriptions = () => {
+    Object.keys(this.events).forEach(correlationId => {
+      this.send({
+        createSubscription: this.events[correlationId].createSubscription,
+        correlationId,
+      })
+
+      // Disable the event listener since we are expecting some initial event messages
+      if (this.events[correlationId]) {
+        this.events[correlationId].listenerEnabled = false
+      }
+
+      // Only enable the event listener after a certain time period, to exclude some initial event messages
+      setTimeout(() => {
+        if (this.events[correlationId]) {
+          this.events[correlationId].listenerEnabled = true
+        }
+      }, this.delayListenersMs)
+    })
+  }
+
+  send = data => {
+    this.ws.send(JSON.stringify(data))
+  }
+
+  subscribe = (createSubscription, correlationId, listener) => {
+    if (this?.ws?.readyState !== 1) {
+      setTimeout(
+        () => this.subscribe(createSubscription, correlationId, listener),
+        this.retrySubscribeMs
+      )
+      return
+    }
+
+    if (!this.events[correlationId]) {
+      // Send a subscription message to the WS
+      this.send({ createSubscription, correlationId })
+      this.events[correlationId] = {
+        createSubscription,
+        listener,
+        listenerEnabled: false,
+      }
+
+      // Only enable the event listener after a certain time period, to exclude some initial event messages
+      setTimeout(() => {
+        if (this.events[correlationId]) {
+          this.events[correlationId].listenerEnabled = true
+        }
+      }, this.delayListenersMs)
+    } else {
+      console.log('already subscribed')
+    }
+  }
+
+  unsubscribe = correlationId => {
+    if (this?.ws?.readyState !== 1) {
+      setTimeout(() => this.unsubscribe(correlationId), this.retrySubscribeMs)
+      return
+    }
+
+    if (this.events[correlationId] && this.idsMap[correlationId]) {
+      // Send an unsubscription message to the WS
+      this.send({
+        cancelSubscription: {
+          subscriptionId: this.idsMap[correlationId],
+        },
+      })
+
+      delete this.events[correlationId]
+      delete this.idsMap[correlationId]
+    }
+  }
+
+  onOpen = () => {}
+
+  onClose = () => {}
+
+  onError = () => {}
+}
+
+export const WebSocketEventClient = new _WebSocketEventClient(
+  '/api/v1/ws/events'
+)
+
+/*

```

Comment:
remove the commented out code?

Reply:
It's an example usage, I made it more clear with another comment.